src/components/Button.js

```diff
@@ -59,38 +61,42 @@ class Button extends React.Component {
   };
 
   render () {
-    // Manually consume everything that isn't going to be passed down to the button so we don't have to keep adding props one at a time.
-    // Keep elementProps for backwards compatibility.
-    const { actionText, buttonRef, children, elementProps, icon, isActive, primaryColor, style, theme, ...rest } = this.props;
-    const mergedTheme = StyleUtils.mergeTheme(theme, primaryColor);
-    const styles = this.styles(mergedTheme);
-
     return (
-      <button
-        className={css({ ...styles.component, ...styles[this.props.type], ...style })}
-        disabled={this.props.type === 'disabled'}
-        ref={buttonRef}
-        {...rest}
-        {...elementProps}
-      >
-        <div style={styles.children}>
-          {(icon && !isActive) && (
-            <Icon
-              size={20}
-              style={styles.icon}
-              type={icon}
-            />
-          )}
-          {isActive && (
-            <Spin direction='counterclockwise'>
-              <Icon size={20} type='spinner' />
-            </Spin>
-          )}
-          <div style={styles.buttonText}>
-            {isActive ? actionText : children}
-          </div>
-        </div>
-      </button>
+      <ThemeContext>{contextTheme => {
+        // Manually consume everything that isn't going to be passed down to the button so we don't have to keep adding props one at a time.
+        // Keep elementProps for backwards compatibility.
+        const { actionText, buttonRef, children, elementProps, icon, isActive, primaryColor, style, theme: propTheme, ...rest } = this.props;
+        const theme = StyleUtils.mergeTheme(contextTheme || propTheme, primaryColor);
```

Comment:
And the merge here would handle if contextTheme was undefined?

Reply:
I just reread this.  Somehow I missed the `||` check you are doing the first time around.  Never mind on this question...